docs/thinking_in_jax.rst

```diff
@@ -0,0 +1,479 @@
+How to think in JAX
```

Comment:
I really like this format instead of notebooks. It should make it easier to comment on PRs and see diffs.

Reply:
Take a look at the update, which uses myst-nb rather than RST. I think you'll like this as well :grin:

Reply:
Update... I decided to punt on myst-nb in this PR, and focus on the content instead. I'll revisit translating our docs to a text-based source in a followup.